addons/mrp/models/mrp_bom.py

```diff
@@ -354,7 +354,7 @@ def _get_default_product_uom_id(self):
         help="BOM Product Variants needed to apply this line.")
     operation_id = fields.Many2one(
         'mrp.routing.workcenter', 'Consumed in Operation', check_company=True,
-        domain="[('routing_id', '=', routing_id), '|', ('company_id', '=', company_id), ('company_id', '=', False)]",
+        domain="[('workcenter_id.active', '=', True), ('routing_id', '=', routing_id), '|', ('company_id', '=', company_id), ('company_id', '=', False)]",
```

Comment:
I think you should rather prevent to archive a work center linked to an operation that is not in an archived routing